internal/auth/transport_test.go

```diff
@@ -22,80 +22,139 @@ func (s roundTripFunc) RoundTrip(r *http.Request) (*http.Response, error) {
+				{
+					Scheme:  "Bearer",
+					TokenFn: func() string { return "static-token" },
+				},
+			},
+			requestContext:  context.Background(),
+			expectedHeaders: []string{"Bearer static-token"},
 		},
 		{
-			name:            "Allow override from context and token provided, and request context unset should use defaultToken",
-			overrideFromCtx: true,
-			staticToken:     "initToken",
-			wrappedTransport: roundTripFunc(func(r *http.Request) (*http.Response, error) {
-				assert.Equal(t, "Bearer initToken", r.Header.Get("Authorization"))
-				return &http.Response{}, nil
-			}),
-			requestContext: context.Background(),
+			name: "Bearer auth with context override - context token used",
+			auths: []Method{
+				{
+					Scheme:  "Bearer",
+					TokenFn: func() string { return "static-token" },
+					FromCtx: bearertoken.GetBearerToken,
+				},
+			},
+			requestContext:  bearertoken.ContextWithBearerToken(context.Background(), "context-token"),
+			expectedHeaders: []string{"Bearer context-token"},
+		},
+		{
+			name: "Multiple auth methods - both tokens added",
+			auths: []Method{
+				{
+					Scheme:  "Bearer",
+					TokenFn: func() string { return "bearer-token" },
+				},
+				{
+					Scheme:  "ApiKey",
+					TokenFn: func() string { return "api-key-value" },
+				},
+			},
+			requestContext:  context.Background(),
+			expectedHeaders: []string{"Bearer bearer-token", "ApiKey api-key-value"},
 		},
 		{
-			name:            "Allow override from context and token provided, and request context set should use context token",
-			overrideFromCtx: true,
-			staticToken:     "initToken",
-			wrappedTransport: roundTripFunc(func(r *http.Request) (*http.Response, error) {
-				assert.Equal(t, "Bearer tokenFromContext", r.Header.Get("Authorization"))
-				return &http.Response{}, nil
-			}),
-			requestContext: bearertoken.ContextWithBearerToken(context.Background(), "tokenFromContext"),
+			name: "Auth config with empty token - no header added",
+			auths: []Method{
+				{
+					Scheme:  "Bearer",
+					TokenFn: func() string { return "" }, // Empty token
+				},
+				{
+					Scheme:  "ApiKey",
+					TokenFn: func() string { return "valid-key" },
+				},
+			},
+			requestContext:  context.Background(),
+			expectedHeaders: []string{"ApiKey valid-key"}, // Only valid token
 		},
 		{
-			name:           "Nil roundTripper provided should return an error",
+			name: "Only context extraction, no context token - no header added",
+			auths: []Method{
+				{
+					Scheme:  "Bearer",
+					FromCtx: bearertoken.GetBearerToken,
+				},
+			},
+			requestContext:     context.Background(),
+			expectNoAuthHeader: true,
+		},
+		{
+			name:           "Nil transport - should return error",
+			auths:          []Method{},
 			requestContext: context.Background(),
-			wantError:      true,
+			expectError:    true,
 		},
-	} {
+	}
+
+	for _, tc := range tests {
 		t.Run(tc.name, func(t *testing.T) {
+			capturedRequest = nil // Reset for each test
```

Comment:
there was no reason to define that variable and transport outside of the for loop - define it here instead, it provides better isolation of the tests (consider adding t.Parallel() to speed them up).